lib/builderator/tasks/packer.rb

```diff
@@ -128,6 +128,42 @@ def wait(profile)
         say_status :complete, 'All copied images are available'
       end
 
+      desc 'remote_tag PROFILE', 'Apply existing tags to the AMI in remote AWS accounts'
+      def remote_tag(profile)
+        invoke :configure, [profile], options
+
+        sts_client = Aws::STS::Client.new(region: Config.aws.region)
+        allowed_cred_keys = %w(access_key_id secret_access_key session_token)
+
+        images.each do |image_name, (image, build)|
+          filters = [{
+            :name => 'name',
+            :values => [image_name]
+          }]
+
+          if build.tagging_role.nil?
+            puts "No remote tagging to be performed as no IAM role is defined"
+            return
+          end
+
+          build.ami_users.each do |account|
+            role_arn = "arn:aws:iam::#{account}:role/#{build.tagging_role}"
+            begin
+              response = sts_client.assume_role( role_arn: role_arn, role_session_name: "tag-new-ami")
+              raise "Could not assume role [#{role_arn}].  Perhaps it does not exist?" unless response.successful?
+            rescue => e
+              puts "=> Got error when trying to assume role:#{e.message} - continuing."
+              next
+            end
+
+            creds_hash = response.credentials.to_h.reject! { |k,v| !allowed_cred_keys.include?(k.to_s) }
```

Comment:
You could use `select!` and implement this in positive terms instead of negating the allowed keys list.